src/placement.c

```diff
@@ -0,0 +1,552 @@
+		 * completely covered by the view. Increment the overlap
+		 * counters these intervals to account for the view.
+		 */
+		for (int i = fr; i < lr; ++i) {
+			for (int j = fc; j < lc; ++j) {
+				overlap_bitmap_index(bmp, i, j) += 1;
+			}
+		}
+	}
+}
+
+/*
+ * Build an overlap bitmap for view->output, which subdivides view->output into
+ * a grid and then counts the number of windows that overlap each grid region.
+ */
+static struct overlap_bitmap
+build_bitmap(struct view *view)
+{
+	assert(view);
+
+	struct overlap_bitmap bitmap = { 0 };
+
+	/* Build the grid edges */
+	build_grid(&bitmap, view);
+	if (bitmap.nr_rows < 1 || bitmap.nr_cols < 1) {
+		return bitmap;
+	}
+
+	/* Determine the grid overlap counts */
+	build_overlap(&bitmap, view);
+
+	return bitmap;
+}
+
+/*
+ * Find the total overlap of an arbitrary region of a given width and height
+ * with intervals in a pre-computed overlap bitmap. The starting interval for
+ * the region is (i, j) in the bitmap grid. If the region is larger than
+ * interval (i, j), neighboring regions will be considered width-wise rightward
+ * (when right is true) or leftward (otherwise) and height-wise downward (when
+ * down is true) or upward (otherwise).
+ *
+ * If the region would extend beyond the edges of the grid (i.e., beyond the
+ * usable region of an output) in the prescribed directions, an overlap of
+ * INT_MAX is returned. Otherwise, the overlap is the sum of the areas of each
+ * interval covered by the region multiplied by its overlap count. For example,
+ * an interval currently covered by three windows will be triply counted in the
+ * overlap sum.
+ */
+static int
+compute_overlap(struct overlap_bitmap *bmp, int i, int j,
+		int width, int height, bool right, bool down, bool *single)
+{
+	/*
+	 * The number of row or column intervals is one less than corresponding
+	 * number of row or column grid points.
+	 */
+	int nri = bmp->nr_rows - 1;
+	int nci = bmp->nr_cols - 1;
+
+	int i_incr = down ? 1 : -1;
+	int j_incr = right ? 1 : -1;
+
+	int overlap = 0;
+	int count = 0;
+
+	/* Walk up or down along rows according to preference */
+	for (int ii = i; ii >= 0 && ii < nri && height > 0; ii += i_incr) {
+		/* Height of this row */
+		int rh = bmp->rows[ii + 1] - bmp->rows[ii];
+
+		/* Height of overlap between this row and test region */
+		int mh = MAX(0, MIN(height, rh));
+
+		/* Remaining height to consider for next row */
+		height -= rh;
+
+		/* Walk left or right along columns according to preference */
+		int ww = width;
+		for (int jj = j; jj >= 0 && jj < nci && ww > 0; jj += j_incr) {
+			/* Width of this column */
+			int cw = bmp->cols[jj + 1] - bmp->cols[jj];
+
+			/* Width of overlap between this column and test region */
+			int mw = MAX(0, MIN(ww, cw));
+
+			/* Add overlap contribution for this interval */
+			overlap += overlap_bitmap_index(bmp, ii, jj) * mh * mw;
+
+			/* Count the number of overlapping intervals */
+			count++;
```

Comment:
It doesn't look like we're using `count`.... unless I'm getting my wires crossed.

Reply:
The `count` variable is used to set the value in `single` before this function returns, which is used to short-circuit the multi-direction check if the view to be placed is not larger than the single grid interval where it starts.

Reply:
Got it. Not sure why I didn't see that.